sfge/src/main/java/com/salesforce/graph/ops/expander/ApexPathExpanderUtil.java

```diff
@@ -67,267 +68,290 @@ private static void logFilteredOutPath(ThrowStatementVertex throwStatementVertex
         }
     }
 
-    /**
-     * The {@code ApexPathCollapser} that keeps track of all paths and attempts to collapse them
-     * whenever a forked method has returned a result.
-     */
-    private final ApexPathCollapser apexPathCollapser;
+    private static class ApexPathExpansionHandler {
+        /** Used to give each object a unique id */
+        private static final AtomicLong ID_GENERATOR = new AtomicLong();
 
-    private ApexPathExpanderUtil(ApexPathExpanderConfig config) {
-        if (config.getDynamicCollapsers().isEmpty()) {
-            this.apexPathCollapser = NoOpApexPathCollapser.getInstance();
-        } else {
-            this.apexPathCollapser = new ApexPathCollapserImpl(config.getDynamicCollapsers());
+        private final Long pathExpansionId;
+
+        private final PathExpansionRegistry registry;
+
+        private ApexPathExpansionHandler(
```

Comment:
Created a new inner class to hold all the instantiation and operations.

Reply:
Question: At this point, might it be better to move these methods back into the outer class and call it `ApexPathExpansionHandler` instead of having an inner class?
We could still have the static `expand()` method to avoid cascading changes.

Reply:
@jfeingold35 There's a static method in this class called `logFilteredOutPath` that all other methods use. To group this the correct way, we probably need a separate class for the static expand() and another separate class for the static logFilteredOutPath. It feels cleaner to leave this the way it is.